File: 11-02-intent-service/app/src/main/java/com/example/intentservice/FirstIntentService.java
package com.example.intentservice;

import android.app.IntentService;
import android.content.Intent;
import android.support.annotation.Nullable;
import android.util.Log;
import android.widget.Toast;

public class FirstIntentService extends IntentService {
    public static final String NUMBER = "NUMBER";
    private int sum;

    public FirstIntentService() {
        super("FirstIntentService");
        sum = 0;
    }

    @Override
    protected void onHandleIntent(@Nullable Intent intent) {
        int number = intent.getIntExtra(NUMBER, 0);
        synchronized (this) {
            try {
                wait(1000);
            } catch (InterruptedException ignored) {
            }
        }
        sum += number;
        Log.d("IntentService", "Sum: " + sum);
    }
}

File: 12-02-intent-service/kotlin-11-02-intent-service /app/src/main/java/com/example/intentservice/FirstIntentService.kt
package com.example.intentservice

import android.app.IntentService
import android.content.Intent
import android.util.Log
import android.widget.Toast

class FirstIntentService : IntentService("FirstIntentService") {
    private var sum: Int = 0

    init {
        sum = 0
    }

    override fun onHandleIntent(intent: Intent?) {
        val number = intent!!.getIntExtra(NUMBER, 0)
        synchronized(this) {
            try {
                Thread.sleep(1000)
            } catch (ignored: InterruptedException) {
            }

        }
        sum += number
        Log.d("IntentService", "Sum: $sum")
    }

    companion object {
        val NUMBER = "NUMBER"
    }
}
